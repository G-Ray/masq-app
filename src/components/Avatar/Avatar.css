.Avatar {
  background-image: url('https://randomuser.me/api/portraits/women/76.jpg');
  width: 120px;
  min-width: 120px;
  height: 120px;
  background-size: cover;
  background-position: top center;
  border-radius: 50%;
  vertical-align: middle;
  display: table-cell;
}
