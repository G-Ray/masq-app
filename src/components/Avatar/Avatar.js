import React from 'react'
import PropTypes from 'prop-types'

import { Camera, AddPictures } from 'icons'

import './Avatar.css'

function Replace (props) {
  return (
    <div className='Avatar upload' style={props.style} onClick={props.onClick}>
      <Camera className='camera' height={33} width={40} color='white' />
    </div>
  )
}

function NewUpload (props) {
  return (
    <AddPictures
      color='var(--grey2-color)'
      className='Avatar upload'
      style={props.style} onClick={props.onClick}
    />
  )
}

export default class Avatar extends React.Component {
  constructor (props) {
    super(props)
    this.openDialog = this.openDialog.bind(this)
  }

  openDialog (e) {
    this.refs.fileDialog.click()
  }

  render () {
    const { image, upload, fileHandler } = this.props
    const style = { backgroundImage: 'url(' + image + ')' }

    return upload
      ? (
        <div>
          <input name='avatar' type='file' ref='fileDialog'
            style={{ display: 'none' }} onChange={fileHandler}
            accept='.jpg, .jpeg, .png'
          />

          {image
            ? <Replace style={style} onClick={this.openDialog} />
            : <NewUpload style={style} onClick={this.openDialog} />
          }
        </div>)
      : <div className='Avatar' style={style} />
  }
}

Avatar.propTypes = {
  upload: PropTypes.bool,
  image: PropTypes.string,
  fileHandler: PropTypes.func
}
