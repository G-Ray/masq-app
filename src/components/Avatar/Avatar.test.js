import React from 'react'
import renderer from 'react-test-renderer'
import Avatar from './Avatar'

it('renders Avatar without upload ability', () => {
  const component = renderer.create(<Avatar />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Avatar with upload ability', () => {
  const component = renderer.create(<Avatar upload />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Avatar with upload ability and an image background', () => {
  const component = renderer.create(<Avatar upload image='https://randomuser.me/api/portraits/women/79.jpg' />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
