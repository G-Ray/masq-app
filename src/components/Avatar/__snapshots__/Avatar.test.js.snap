// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Avatar with upload ability 1`] = `
<div>
  <input
    accept=".jpg, .jpeg, .png"
    name="avatar"
    onChange={undefined}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
  <svg
    className="Avatar upload"
    color="var(--grey2-color)"
    height={120}
    onClick={[Function]}
    style={
      Object {
        "backgroundImage": "url(undefined)",
      }
    }
    viewBox="0 0 120 120"
    width={120}
  >
    <g
      fill="none"
      fillRule="evenodd"
    >
      <circle
        cx={60}
        cy={60}
        fill="#fff"
        r={60}
      />
      <circle
        cx={60}
        cy={60}
        r={59}
        stroke="currentColor"
        strokeWidth={2}
      />
      <path
        d="M76.8 46.933h-6.533l-3.174-4.853c-.373-.373-.933-.747-1.493-.747H54.4c-.56 0-1.12.374-1.493.747l-3.174 4.853H43.2c-3.173 0-5.6 2.427-5.6 5.6v20.534c0 3.173 2.427 5.6 5.6 5.6h33.6c3.173 0 5.6-2.427 5.6-5.6V52.533c0-3.173-2.427-5.6-5.6-5.6zm1.867 26.134c0 1.12-.747 1.866-1.867 1.866H43.2c-1.12 0-1.867-.746-1.867-1.866V52.533c0-1.12.747-1.866 1.867-1.866h7.467c.56 0 1.12-.374 1.493-.747l3.173-4.853h9.147l3.173 4.853c.56.373 1.12.747 1.68.747H76.8c1.12 0 1.867.746 1.867 1.866zM60 52.533c-5.227 0-9.333 4.107-9.333 9.334 0 5.226 4.106 9.333 9.333 9.333s9.333-4.107 9.333-9.333c0-5.227-4.106-9.334-9.333-9.334zm0 14.934c-3.173 0-5.6-2.427-5.6-5.6 0-3.174 2.427-5.6 5.6-5.6s5.6 2.426 5.6 5.6c0 3.173-2.427 5.6-5.6 5.6z"
        fill="currentColor"
        fillRule="nonzero"
      />
    </g>
  </svg>
</div>
`;

exports[`renders Avatar with upload ability and an image background 1`] = `
<div>
  <input
    accept=".jpg, .jpeg, .png"
    name="avatar"
    onChange={undefined}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
  <div
    className="Avatar upload"
    onClick={[Function]}
    style={
      Object {
        "backgroundImage": "url(https://randomuser.me/api/portraits/women/79.jpg)",
      }
    }
  >
    <svg
      className="camera"
      color="white"
      height={33}
      viewBox="0 0 24 20"
      width={40}
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <defs>
        <path
          d="M21 3h-3.5L15.8.4c-.2-.2-.5-.4-.8-.4H9c-.3 0-.6.2-.8.4L6.5 3H3C1.3 3 0 4.3 0 6v11c0 1.7 1.3 3 3 3h18c1.7 0 3-1.3 3-3V6c0-1.7-1.3-3-3-3zm1 14c0 .6-.4 1-1 1H3c-.6 0-1-.4-1-1V6c0-.6.4-1 1-1h4c.3 0 .6-.2.8-.4L9.5 2h4.9l1.7 2.6c.3.2.6.4.9.4h4c.6 0 1 .4 1 1v11zM12 6c-2.8 0-5 2.2-5 5s2.2 5 5 5 5-2.2 5-5-2.2-5-5-5zm0 8c-1.7 0-3-1.3-3-3s1.3-3 3-3 3 1.3 3 3-1.3 3-3 3z"
          id="camera-svg"
        />
      </defs>
      <use
        fill="currentColor"
        fillRule="nonzero"
        xlinkHref="#camera-svg"
      />
    </svg>
  </div>
</div>
`;

exports[`renders Avatar without upload ability 1`] = `
<div
  className="Avatar"
  style={
    Object {
      "backgroundImage": "url(undefined)",
    }
  }
/>
`;
